sky/provision/kubernetes/utils.py

```diff
@@ -170,7 +179,7 @@ def wrapper(*args, **kwargs):
                 error_msg = (f'Kubernetes configuration error{resource_msg}: '
                              f'{str(last_exception)}')
 
-            raise exceptions.ResourcesUnavailableError(
+            raise exceptions.KubeAPIUnreachableError(
                 f'{error_msg}'
                 f' Please check if the cluster is healthy and retry.'
                 f'{debug_cmd}') from last_exception
```

Comment:
Need to double check all callers of these functions that will raise the new exception, to make sure that the failover will not be interrupted by this error, especially for the code in managed jobs.

Reply:
Assuming the failover works previously, then existing callers should handle the parent exception `ResourcesUnavailableError` properly, thus covering `KubeAPIUnreachableError`. 

The failover smoke test case is extended to cover the failover during optimizing. Provision stage failover is a bit tricky to cover in smoke test, since we need to mock a cluster that reachable during optimizing but unreachable during provisioning. Do you have any suggestion?